src/lightning/pytorch/loops/evaluation_loop.py

```diff
@@ -56,7 +56,8 @@ def __init__(self, trainer: "pl.Trainer", verbose: bool = True, inference_mode:
         self.verbose = verbose
         self.inference_mode = inference_mode
         self.batch_progress = _BatchProgress()  # across dataloaders
-        self._max_batches: List[Union[int, float]] = []
+        #  list in "sequential" mode, number otherwise
+        self._max_batches: Union[int, float, List[Union[int, float]]] = []
```

Comment:
cc @justusschock @awaelchli for some early feedback:

When "sequential" mode is not used, do you think it should return a number (as currently implemented, matching what FitLoop returns), or a list of size 1?

Reply:
Probably the same as in fitloop for consistency, and then later one could think about refactoring both if having always a list makes the code cleaner?

Reply:
Okay. I kept the same structure. The impact of this is having to check whether the property returns a list.

Refactoring later would be nice, but it would be a breaking change unless we add new properties that have the new behaviour.